appinventor/buildserver/src/com/google/appinventor/buildserver/resources/runtime.scm

```diff
@@ -407,7 +408,8 @@
            ;; for event handlers and component names.
            (let ((registeredObject (string->symbol registeredComponentName)))
                  (if (is-bound-in-form-environment registeredObject)
-                     (if (eq? (lookup-in-form-environment registeredObject) componentObject)
+                     ;; temporarily deleted because I have no ideas how to get true out of this condition using isMulti
+                     ;;(if (or (eq? (lookup-in-form-environment registeredObject) componentObject) (isMulti))
```

Comment:
Check with Mark Friedman.   I don't understand the implications of what you
are doing here.


== Hal Abelson
hal@mit.edu
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.
MIT Media Lab, Second Class Faculty


On Thu, Apr 2, 2015 at 8:13 PM, gitars <notifications@github.com> wrote:

> In
> appinventor/buildserver/src/com/google/appinventor/buildserver/resources/runtime.scm
> <https://github.com/mit-cml/appinventor-sources/pull/347#discussion_r27708637>
> :
>
> > @@ -407,7 +408,8 @@
> >             ;; for event handlers and component names.
> >             (let ((registeredObject (string->symbol registeredComponentName)))
> >                   (if (is-bound-in-form-environment registeredObject)
> > -                     (if (eq? (lookup-in-form-environment registeredObject) componentObject)
> > +                     ;; temporarily deleted because I have no ideas how to get true out of this condition using isMulti
> > +                     ;;(if (or (eq? (lookup-in-form-environment registeredObject) componentObject) (isMulti))
>
> So I have no idea how to write code here, everything I tried for an hour
> failed. Basically, I know we do this check for a reason, but because of the
> way I implemented multi-generic events, I want to ignore it if it such an
> event. So I passed in a parameter isMulti, and if it is true, I want to
> pass this condition. Every way I've tried to do this (by using different
> types for isMulti such as java.lang.Boolean/java.lang.String and comparing
> it to thing failed).
>
> However, the multi-generic event handler seems to work here if I simply
> ignore the condition, so if we get the condition right, it should still
> work.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-cml/appinventor-sources/pull/347/files#r27708637>.
>